### ISSUE TITLE
Reported F1 in your blog are far from what I get running your code

### ISSUE THREAD

**Issue by monajalal** [vipzgy/AttentionTargetSentiment#6]
Hey
Running your code I get the following F1s:
[jalal@goku AttentionTargetSentiment]$ python main.py -cuda -getF1 -which-model contextualizedgates  -batch-size 1

Loading data...

Parameters:
	ATTENTION_SIZE=150
	BATCH_SIZE=1
	CLIP_NORM=None
	CUDA=True
	DEVICE=-1
	DROPOUT_EMBED=0.2
	DROPOUT_RNN=0.4
	EMBED_DIM=200
	EMBED_NUM=23095
	EPOCHS=30
	GETF1=True
	GRAYSCALE=None
	HIDDEN_SIZE=150
	IF_RE=False
	LABEL_NUM=3
	LOG_INTERVAL=1
	LR=0.01
	LR_SCHEDULER=None
	MAX_NORM=None
	MESSAGE=tt
	NEED_SMALLEMBED=False
	SAVE_DIR=snapshot/tt
	SAVE_INTERVAL=100
	SHUFFLE=True
	SNAPSHOT=None
	TEST=False
	TEST_INTERVAL=100
	USE_EMBEDDING=True
	WEIGHT_DECAY=1e-06
	WHICH_DATA=Z
	WHICH_EMBEDDING=200d
	WHICH_INIT=xavier
	WHICH_MODEL=contextualizedgates
	WHICH_OPTIM=Adagrad
/scratch2/debate_tweets/sentiment/AttentionTargetSentiment/model/attention.py:28: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  alpha = F.softmax(beta)
/scratch2/debate_tweets/sentiment/AttentionTargetSentiment/model/contextualizedgates.py:130: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  z_all = F.softmax(z_all)

polarity: neutral  corrects: 1  predicts: 3  size: 581
polarity: neutral  recall: 0.1721%  precision: 33.3333%  f1: 0.3425% 

polarity: positive  corrects: 291  predicts: 1163  size: 294
polarity: positive  recall: 98.9796%  precision: 25.0215%  f1: 39.9451% 

polarity: negative  corrects: 1  predicts: 4  size: 295
polarity: negative  recall: 0.3390%  precision: 25.0000%  f1: 0.6689% 

13.652151576879215
[jalal@goku AttentionTargetSentiment]$ python main.py -cuda -getF1 -which-model contextualizedgates -which-embedding 200dt  -batch-size 1

Loading data...

Parameters:
	ATTENTION_SIZE=150
	BATCH_SIZE=1
	CLIP_NORM=None
	CUDA=True
	DEVICE=-1
	DROPOUT_EMBED=0.2
	DROPOUT_RNN=0.4
	EMBED_DIM=200
	EMBED_NUM=23095
	EPOCHS=30
	GETF1=True
	GRAYSCALE=None
	HIDDEN_SIZE=150
	IF_RE=False
	LABEL_NUM=3
	LOG_INTERVAL=1
	LR=0.01
	LR_SCHEDULER=None
	MAX_NORM=None
	MESSAGE=tt
	NEED_SMALLEMBED=False
	SAVE_DIR=snapshot/tt
	SAVE_INTERVAL=100
	SHUFFLE=True
	SNAPSHOT=None
	TEST=False
	TEST_INTERVAL=100
	USE_EMBEDDING=True
	WEIGHT_DECAY=1e-06
	WHICH_DATA=Z
	WHICH_EMBEDDING=200dt
	WHICH_INIT=xavier
	WHICH_MODEL=contextualizedgates
	WHICH_OPTIM=Adagrad
/scratch2/debate_tweets/sentiment/AttentionTargetSentiment/model/attention.py:28: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  alpha = F.softmax(beta)
/scratch2/debate_tweets/sentiment/AttentionTargetSentiment/model/contextualizedgates.py:130: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  z_all = F.softmax(z_all)

polarity: neutral  corrects: 16  predicts: 25  size: 581
polarity: neutral  recall: 2.7539%  precision: 64.0000%  f1: 5.2805% 

polarity: positive  corrects: 277  predicts: 1089  size: 294
polarity: positive  recall: 94.2177%  precision: 25.4362%  f1: 40.0578% 

polarity: negative  corrects: 15  predicts: 56  size: 295
polarity: negative  recall: 5.0847%  precision: 26.7857%  f1: 8.5470% 

17.961793954577615
[jalal@goku AttentionTargetSentiment]$ 

Which is very different from what you have reported here for contextualized gates model. Can you please show that with which parameters you run the model?
![screenshot from 2018-04-23 01-21-27](https://user-images.githubusercontent.com/1892917/39108140-b177b548-4694-11e8-8fea-d3870c16c358.png)

https://www.zybuluo.com/habits/note/847972#2contextualized-attention-modelbilstm-att-c
